src/views/v2/list/top-bar/datepicker.php

```diff
@@ -51,19 +53,29 @@
 ?>
 <div class="tribe-events-c-top-bar__datepicker">
 	<button
-		class="tribe-common-h2 tribe-common-h3--min-medium tribe-common-h--alt tribe-events-c-top-bar__datepicker-button"
+		class="tribe-common-h3 tribe-common-h--alt tribe-events-c-top-bar__datepicker-button"
 		data-js="tribe-events-top-bar-datepicker-button"
 	>
 		<?php if ( $is_now ) : ?>
 			<?php esc_html_e( 'Now', 'the-events-calendar' ); ?>
 		<?php else: ?>
 			<time datetime="<?php echo esc_attr( date_i18n( 'Y-m-d', $selected_start_datetime ) ); ?>">
-				<?php echo esc_html( $selected_start_date_label ); ?>
+				<span class="tribe-events-c-top-bar__datepicker-date-mobile">
+					<?php echo Dates::build_date_object( $selected_start_datetime )->format( $date_compact_format ); ?>
+				</span>
+				<span class="tribe-events-c-top-bar__datepicker-date-desktop tribe-common-a11y-hidden">
+					<?php echo esc_html( $selected_start_date_label ); ?>
+				</span>
 			</time>
 		<?php endif; ?>
 		&mdash;
 		<time datetime="<?php echo esc_attr( date_i18n( 'Y-m-d', $selected_end_datetime ) ); ?>">
-			<?php echo esc_html( $selected_end_date_label ); ?>
+			<span class="tribe-events-c-top-bar__datepicker-date-mobile">
+				<?php echo Dates::build_date_object( $selected_end_datetime )->format( $date_compact_format ); ?>
```

Comment:
can you build this above first, and echo the var instead of building it on the spot? also missing `esc_html()`

Reply:
Not sure if esc_html should apply here. We're printing a date, formatted with our own formats, and we're certain that they don't contain HTML. https://cloudup.com/cphJrxi-OkM